package SensorDemo;

message Message {
  enum MessageType {
    INIT = 0;
    SENSOR_DATA = 1;
    OTHER = 2;
  }

  enum MessageEncAlgorithm {
    AES = 0;
    DES = 1;
    DES3 = 2;
    NONE = 3;
  }

  enum MessageIbeAlgorithm {
    WATERS05 = 0;
    WATERS09 = 1;
  }

  required MessageType type = 1 [default = SENSOR_DATA];
  required MessageEncAlgorithm encAlgorithm = 2 [default = AES];
  required MessageIbeAlgorithm ibeAlgorithm = 3 [default = WATERS09];
  optional string identityBasedMasterPublicKey = 4;
  optional string identityBasedEncryptedKey = 5;
  optional string encryptedMessage = 6;
  required string nonce = 7;
  required int32 timestamp = 8;
}

